src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_comm.py

```diff
@@ -404,18 +413,31 @@ def do_kill_pydev_thread(self):
 
 
 def create_server_socket(host, port):
-    s = socket(AF_INET, SOCK_STREAM)
-    s.settimeout(None)
+    server = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)
+    if IS_WINDOWS:
+        try:
+            from socket import SIO_LOOPBACK_FAST_PATH
+        except ImportError:
+            pass # Not supported in python 2.* or <3.6
+        else:
+            try:
+                server.ioctl(SIO_LOOPBACK_FAST_PATH, True)
+            except OSError as ose:
+                if ose.winerror != 10045:  # Not supported by OS
+                    raise
+        from socket import SO_EXCLUSIVEADDRUSE
+        server.setsockopt(SOL_SOCKET, SO_EXCLUSIVEADDRUSE, 1)
+    else:
+        server.setsockopt(SOL_SOCKET, SO_REUSEADDR, 1)
 
     try:
-        from socket import SO_REUSEPORT
-        s.setsockopt(SOL_SOCKET, SO_REUSEPORT, 1)
-    except ImportError:
-        s.setsockopt(SOL_SOCKET, SO_REUSEADDR, 1)
+        server.bind((host, port))
+        server.settimeout(None)
+    except Exception:
+        server.close()
+        raise
```

Comment:
Didn't notice this one earlier, but I think this should also do cleanup for all the calls on the socket above - e.g. `ioctl` can definitely fail, and I'm not sure, but `setsockopt` might also fail? There's definitely that line above that will re-raise exception from `ioctl` which will result in the socket not getting cleaned up.

It's probably best to wrap the entire function body after `server` is created into one giant `try..except` that closes and re-raises.